components/root/root.tsx

```diff
@@ -92,7 +98,7 @@ const Authorize = makeAsyncComponent('Authorize', LazyAuthorize);
 const Mfa = makeAsyncComponent('Mfa', LazyMfa);
 const PreparingWorkspace = makeAsyncComponent('PreparingWorkspace', LazyPreparingWorkspace);
 
-const LoggedInRoute = ({component: Component, ...rest}) => (
+const LoggedInRoute = ({component: Component, ...rest}: {component: any; path: string}) => (
```

Comment:
Closest I could get was doing something similar, but pretending that the route props are the component props that were wanted. If I don't do that (the `as unknown as T`, then I don't for sure know whether the component wanted the route props, and I get a type error). What do you think?

Reply:
I think this is probably fine as-is then. I'm pretty sure we're doing some stuff with these components that makes this typing more difficult than it needs to be. I tried a few different combinations of things, but I couldn't get everything to play nicely myself. `RouteComponentProps` might need to take part of `T` as a parameter containing the query params that `Component` wants, but I couldn't manage to sort that out entirely.

I also noticed a few things that seemed off, but they're not from this PR, so I don't think it's necessary to actually fix them here:
1. `PreparingWorkspace` seems to require an extra `handleForm` prop that's not actually used
2. We're passing the route props from `LoggedInRoute` to `NeedsTeam`, but `NeedsTeam` also uses the `useRouter` HOC, so there might be some conflicts there
3. We pass the same props into `LoggedIn` as we do `Component`, so that might be causing something funny to happen